src/bench/crypto_hash.cpp

```diff
@@ -75,6 +75,26 @@ static void SipHash_32b(benchmark::State& state)
     }
 }
 
+static void SipHash_3b(benchmark::State& state)
+{
+    uint64_t hash = 0;
+    uint64_t k2 = 0;
+    std::vector<uint8_t> in(3, 0xee);
+    while (state.KeepRunning()) {
+        hash = CSipHasher(hash, ++k2).Write(in.data(), in.size()).Finalize();
+    }
+}
+
+
+static void SipHash(benchmark::State& state)
```

Comment:
nit: I'm not that much into crypto but `SipHash` and SipHash_32b look very similar. Maybe you could include them into one function or make one general function which is called by the both. The only difference between them is the vector constructor

Reply:
`SipHash_32b` is calling a different SipHash implementation that is somewhat optimized to 32 bytes (256bit)